engine/schema/src/main/java/com/cloud/usage/dao/UsageDaoImpl.java

```diff
@@ -483,6 +483,8 @@ public void removeOldUsageRecords(int days) {
             s_logger.error("error removing old cloud_usage records for interval: " + days);
         } finally {
             txn.close();
+            TransactionLegacy swap = TransactionLegacy.open(TransactionLegacy.CLOUD_DB);
```

Comment:
I'm also wondering if there's another way to do this, like wrap in a Transaction::execute? LGTM otherwise